From 871b0ed1847c3c5413a4ca72ecf18735858f7708 Mon Sep 17 00:00:00 2001
From: Josh Boyer <jwboyer@fedoraproject.org>
Date: Tue, 27 Aug 2013 13:33:03 -0400
Subject: [PATCH 14/14] efi: Add EFI_SECURE_BOOT bit

Patch-mainline: Not yet, fedora 20
References: fate#314486
Target: SLE-12

UEFI machines can be booted in Secure Boot mode.  Add a EFI_SECURE_BOOT bit
for use with efi_enabled.

Signed-off-by: Josh Boyer <jwboyer@fedoraproject.org>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

---
 arch/x86/kernel/setup.c |    2 ++
 include/linux/efi.h     |    1 +
 2 files changed, 3 insertions(+)

--- a/arch/x86/kernel/setup.c
+++ b/arch/x86/kernel/setup.c
@@ -1146,7 +1146,9 @@ void __init setup_arch(char **cmdline_p)
 
 #ifdef CONFIG_EFI_SECURE_BOOT_SECURELEVEL
 	if (boot_params.secure_boot) {
+		set_bit(EFI_SECURE_BOOT, &efi.flags);
 		set_securelevel(1);
+		pr_info("Secure boot enabled\n");
 	}
 #endif
 
--- a/include/linux/efi.h
+++ b/include/linux/efi.h
@@ -1005,6 +1005,7 @@ extern int __init efi_setup_pcdp_console
 #define EFI_ARCH_1		7	/* First arch-specific bit */
 #define EFI_DBG			8	/* Print additional debug info at runtime */
 #define EFI_NX_PE_DATA		9	/* Can runtime data regions be mapped non-executable? */
+#define EFI_SECURE_BOOT		10 	/* Are we in Secure Boot mode? */
 
 #ifdef CONFIG_EFI
 /*
